tests/api2/test_bootenv.py

```diff
@@ -1,24 +1,68 @@
+from functions import wait_on_job
 from middlewared.test.integration.utils import call, ssh
 
+def test_get_default_environment_and_make_new_one():
+    active_be_id = call('bootenv.query', [['activated', '=', True]], {'get': True})['id']
+
+    # create duplicate name to test failure
+    results = call('bootenv.create', {'name': active_be_id, 'source': active_be_id})
+    assert results == f'[EEXIST] bootenv_create.name: The name "{active_be_id}" already exists'
+
+    # create new bootenv and activate it
+    call('bootenv.create', {'name': 'bootenv01', 'source': active_be_id})
+    assert len(call('bootenv.query', [['name', '=', 'bootenv01']]).json()) == 1
+    call('bootenv.activate', 'bootenv01')
+
+
+# Update tests
+def test_cloning_a_new_boot_environment():
+    call('bootenv.create', {'name': 'bootenv02', 'source': 'bootenv01'})
+    call('bootenv.activate', 'bootenv02')
+
+
+def test_change_boot_environment_name_and_attributes():
+    call('bootenv.update', 'bootenv01', {'name': 'bootenv03'})
+    call('bootenv.set_attribute', 'bootenv03', {'keep': True})
+    call('bootenv.activate', 'bootenv03')
+
+
+# Delete tests
+def test_removing_a_boot_environment_02():
+    job_id = call('bootenv.delete', 'bootenv02')
+    job_status = wait_on_job(job_id, 180)
+    assert job_status['state'] == 'SUCCESS'
+
+
+def test_activate_original_bootenv():
+    be_id = call('bootenv.query', [['name', '!=', 'bootenv03']], {'get': True})["id"]
+    call('bootenv.activate', be_id)
+
+
+def test_removing_a_boot_environment_03():
+    call('bootenv.set_attribute', 'bootenv03', {'keep': False})
+    job_id = call('bootenv.delete', 'bootenv03')
+    job_status = wait_on_job(job_id, 180)
```

Comment:
`wait_on_job` isn't needed when using our websocket client. `job=True` can be passed to the client and it'll wait automatically.